server/src/main/java/org/elasticsearch/monitor/jvm/HotThreads.java

```diff
@@ -104,6 +105,34 @@ public static void logLocalHotThreads(Logger logger, Level level, String prefix,
         }
     }
 
+    /**
+     * Capture and log the current threads on the local node. Unlike hot threads this does nto sample and captures current state only.
+     * Useful for capturing stack traces for unexpectedly-slow operations in production. The resulting message might be large so it is
+     * split per thread and logged as multiple entries.
+     *
+     * @param logger        The logger to use for the logging
+     * @param level         The log level to use for the logging.
+     * @param prefix        The prefix to emit on each chunk of the logging.
+     * @param referenceDocs A link to the docs describing how to decode the logging.
+     */
+    public static void logLocalCurrentThreads(Logger logger, Level level, String prefix, ReferenceDocs referenceDocs) {
+        if (logger.isEnabled(level) == false) {
+            return;
+        }
+
+        try (var writer = new StringWriter()) {
+            new HotThreads().busiestThreads(500).threadElementsSnapshotCount(1).ignoreIdleThreads(false).detect(writer, () -> {
+                logger.log(level, "{}: {}", prefix, writer.toString(), referenceDocs);
```

Comment:
Are we ignoring referenceDocs here? The message has only two placeholders. I do think logging referenceDocs for every thread is excessive. But I am not sure the intention here.

Reply:
The point of the `ReferenceDocs` link was largely so that we had somewhere to write instructions about how to decode these stack traces. Not really necessary if we're not encoding them in the first place.